pkg/kp/pcstore/consul_store.go

```diff
@@ -272,6 +255,62 @@ func (s *consulStore) Watch(id fields.ID, quit <-chan struct{}) <-chan WatchedPo
 	return outCh
 }
 
+// WatchPodCluster implements a watch for the Pod cluster at _id_
+// It will return a blocking channel on which the client can read
+// WatchedPodCluster objects. The goroutine maintaining the watch will block on
+// writing to this channel so it's up to the caller to read it with haste.
+func (s *consulStore) WatchPodCluster(id fields.ID, quit <-chan struct{}) <-chan WatchedPodCluster {
+	inCh := make(chan *api.KVPair)
+	outCh := make(chan WatchedPodCluster)
+	errChan := make(chan error, 1)
+	key := path.Join(podClusterTree, string(id))
+
+	go func() {
+		for {
+			select {
+			case <-quit:
+				return
+			case err := <-errChan:
+				s.logger.WithError(err).Errorf("WatchPrefix returned error, recovered.")
+			}
+		}
+	}()
+
+	go consulutil.WatchSingle(key, s.kv, inCh, quit, errChan)
+
+	go func() {
+		var kvp *api.KVPair
+		for {
+			select {
+			case kvp = <-inCh:
+				if kvp == nil {
+					// PodCluster at id has been deleted
```

Comment:
Seems reasonable to me. So this `return` makes it so that the goroutine watching `inCh` will stop. Is there something to stop the one watching `errChan` (and should there be?)?